specification/cognitiveservices/data-plane/AzureOpenAI/authoring/stable/2022-12-01/azureopenai.json

```diff
@@ -0,0 +1,1965 @@
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Deleting a deployment.": {
+            "$ref": "./examples/delete_deployment.json"
+          }
+        }
+      }
+    },
+    "/files": {
+      "get": {
+        "tags": [
+          "Files:"
+        ],
+        "summary": "Gets a list of all files owned by the Azure OpenAI resource.\r\nThese include user uploaded content like files with purpose \"fine-tune\" for training or validation of fine-tunes models\r\nas well as files that are generated by the service such as \"fine-tune-results\" which contains various metrics for the corresponding fine-tune job.",
+        "operationId": "Files_List",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionQueryParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/FileList"
+            }
+          },
+          "default": {
+            "description": "An error occurred.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Getting all files owned by this account.": {
+            "$ref": "./examples/get_files.json"
+          }
+        }
+      },
+      "post": {
+        "tags": [
+          "Files:"
+        ],
+        "summary": "Creates a new file entity by uploading data from a local machine. Uploaded files can, for example, be used for training or evaluating fine-tuned models.",
+        "operationId": "Files_Upload",
+        "consumes": [
+          "multipart/form-data"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionQueryParameter"
+          },
+          {
+            "in": "formData",
+            "name": "purpose",
+            "description": "The intended purpose of the uploaded documents. Use \"fine-tune\" for fine-tuning. This allows us to validate the format of the uploaded file.",
+            "required": true,
+            "type": "string",
+            "enum": [
+              "fine-tune",
+              "fine-tune-results"
+            ],
+            "x-ms-enum": {
+              "name": "Purpose",
+              "modelAsString": true,
+              "values": [
+                {
+                  "value": "fine-tune",
+                  "description": "This file contains training data for a fine tune job."
+                },
+                {
+                  "value": "fine-tune-results",
+                  "description": "This file contains the results of a fine tune job."
+                }
+              ]
+            }
+          },
```

Comment:
If you changed this parameter to a query parameter could you change the operation to accept an application/octet-stream body rather than multipart/form-data? If so that would probably be easier for users.

Reply:
In order to reduce the amount of differences to the minimum we prefer to stick with the OpenAI design of this method. 

We added the import method to support larger files and that one does not have this problem.